internal/app/plugins/sqs/sqs.go

```diff
@@ -0,0 +1,168 @@
+	Timeout time.Duration `flag:"timeout" desc:"aws request timeout" default:"30s"`
+}
+
+type SQSClient interface {
+	GetQueueUrl(ctx context.Context, params *sqs.GetQueueUrlInput, optFns ...func(*sqs.Options)) (*sqs.GetQueueUrlOutput, error)
+	SendMessage(ctx context.Context, params *sqs.SendMessageInput, optFns ...func(*sqs.Options)) (*sqs.SendMessageOutput, error)
+}
+
+type SQSWorker struct {
+	i       int
+	sq      <-chan *aio.Message
+	timeout time.Duration
+	aio     aio.AIO
+	metrics *metrics.Metrics
+	client  SQSClient
+}
+
+type SQS struct {
+	sq      chan *aio.Message
+	workers []*SQSWorker
+}
+
+type Addr struct {
+	Region string `json:"region"`
+	Queue  string `json:"queue"`
+}
+
+func New(a aio.AIO, metrics *metrics.Metrics, config *Config) (*SQS, error) {
+	return NewWithClient(a, metrics, config, nil)
+}
+
+func NewWithClient(a aio.AIO, metrics *metrics.Metrics, config *Config, client SQSClient) (*SQS, error) {
+	sq := make(chan *aio.Message, config.Size)
+	workers := make([]*SQSWorker, config.Workers)
+
+	for i := 0; i < config.Workers; i++ {
+		workers[i] = &SQSWorker{
+			i:       i,
+			sq:      sq,
+			timeout: config.Timeout,
+			aio:     a,
+			metrics: metrics,
+			client:  client,
+		}
+	}
+
+	return &SQS{
+		sq:      sq,
+		workers: workers,
+	}, nil
+}
+
+func (s *SQS) String() string {
+	return fmt.Sprintf("%s:sqs", t_aio.Sender.String())
+}
+
+func (s *SQS) Type() string {
+	return "sqs"
+}
+
+func (s *SQS) Start(chan<- error) error {
+	for _, worker := range s.workers {
+		go worker.Start()
+	}
+
+	return nil
+}
+
+func (s *SQS) Stop() error {
+	close(s.sq)
+	return nil
+}
+
+func (s *SQS) Enqueue(msg *aio.Message) bool {
+	select {
+	case s.sq <- msg:
+		return true
+	default:
+		return false
+	}
+}
+
+func (w *SQSWorker) String() string {
+	return fmt.Sprintf("%s:sqs", t_aio.Sender.String())
+}
+
+func (w *SQSWorker) Start() {
+	counter := w.metrics.AioWorkerInFlight.WithLabelValues(w.String(), strconv.Itoa(w.i))
+	w.metrics.AioWorker.WithLabelValues(w.String()).Inc()
+	defer w.metrics.AioWorker.WithLabelValues(w.String()).Dec()
+
+	for {
+		msg, ok := <-w.sq
+		if !ok {
+			return
+		}
+
+		counter.Inc()
+		msg.Done(w.Process(msg.Addr, msg.Body))
+		counter.Dec()
+	}
+}
+
+func (w *SQSWorker) Process(data []byte, body []byte) (bool, error) {
+	var addr *Addr
+	if err := json.Unmarshal(data, &addr); err != nil {
+		return false, err
+	}
+
+	if addr.Region == "" {
+		return false, errors.New("missing region in address")
+	}
+
+	if addr.Queue == "" {
+		return false, errors.New("missing queue in address")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), w.timeout)
+	defer cancel()
+
+	client := w.client
+	if client == nil {
+		cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(addr.Region))
+		if err != nil {
+			return false, fmt.Errorf("failed to load AWS config: %w", err)
+		}
+		client = sqs.NewFromConfig(cfg)
+	}
+
+	q_url, err := client.GetQueueUrl(ctx, &sqs.GetQueueUrlInput{
```

Comment:
Alright this is a little out there, but bear with me.

What if, instead of providing an sqs url that looks like this:
```
sqs://us-west-2/my-queue
```

We have users provide a url that looks liks this:
```
sqs+https://sqs.us-west-2.amazonaws.com/0123456789012/my-queue
```

This certainly looks odd, but it has two advantages:
1. we avoid one network request
2. the queue url is easy to copy in the aws console ui

I checked golangs url parser and this is indeed considered a valid URL so it should be pretty straightforward to parse the relvant bit (https://sqs.us-west-2.amazonaws.com/0123456789012/my-queue) in the sender.